### PR TITLE
Feat/profiling/android

### PR DESCRIPTION
**Pull request by stefanosiano**
## :scroll: Description
Add profiling feature to Sentry v6

#skip-changelog

### CODE REVIEW

**Comment by stefanosiano** [comment]
Review not required to merge... 🤔 😈

**Comment by marandaneto** [comment]
@stefanosiano can you point us to which commits actually implement the serialization/deserialization of the new protocol? otherwise, it's hard to review.

**Comment by marandaneto** [comment]
> Review not required to merge... 🤔 😈

Such rules exist only in `main`

**Comment by stefanosiano** [comment]
@marandaneto Sure, the json commit is https://github.com/getsentry/sentry-java/pull/1973/commits/c860b8c1e83fe446873924a0868a819e1d0aea85
The other commit for v6 is https://github.com/getsentry/sentry-java/pull/1973/commits/5732ed18d101c4576d93688a9403aa32bc3c45ac, which contains very minor changes to make it compatible with v6 (e.g. `IBuildInfoProvider` -> `BuildInfoProvider`)

**Comment by stefanosiano** [comment]
@marandaneto Can i merge it or should i wait for your review?

**Comment by marandaneto** [comment]
> @marandaneto Can i merge it or should i wait for your review?

@romtsn would you like to take this one?

**Comment by romtsn** [comment]
@marandaneto  will do!

**Review by romtsn — APPROVED** [review]
Few minor comments, but lgtm otherwise!

File: Sentry/src/test/java/io/sentry/NoOpTransactionProfilerTest.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package io.sentry
```
[the remark this answers is not in the document]
Comment (stefanosiano) [reply Sentry/src/test/java/io/sentry/NoOpTransactionProfilerTest.kt:1]: This is weird... there's no Sentry folder anywhere... maybe due to mac being case insensitive.
However, I fixed it with command line :/

**Pull request merged**